### ISSUE TITLE
Docker is not creating sub-directories

### ISSUE THREAD

**Issue by espritengineer** [docker/docker#24673]
**Output of `docker version`:**

```
Client:
 Version:      1.11.2
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   b9f10c9
 Built:        Wed Jun  1 21:23:11 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.2
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   b9f10c9
 Built:        Wed Jun  1 21:23:11 2016
 OS/Arch:      linux/amd64

```

**Output of `docker info`:**

```
Containers: 4
 Running: 3
 Paused: 0
 Stopped: 1
Images: 34
Server Version: 1.11.2
Storage Driver: devicemapper
 Pool Name: docker-202:2-426670-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 2.004 GB
 Data Space Total: 107.4 GB
 Data Space Available: 22.06 GB
 Metadata Space Used: 4.489 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.143 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Either use `--storage-opt dm.thinpooldev` or use `--storage-opt dm.no_warn_on_loop_devices=true` to suppress this warning.
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.107-RHEL7 (2016-06-09)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: host bridge null
Kernel Version: 3.10.0-327.22.2.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 11.33 GiB
Name: internal.roam-smart.com
ID: 6BF2:BEN6:RVCK:Z2HV:SSI3:VEFG:N4H4:ZDRX:CVQ4:25JN:IENW:TF4R
Docker Root Dir: /var/lib/docker
Debug mode (client): false
Debug mode (server): false
Registry: https://index.docker.io/v1/
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled

```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Virtual machine on IBM Softlayer cloud

**Steps to reproduce the issue:**
1. Created a Dockerfile to build an image
2. Added the following instructions:
ENV JVM_PATH /acme/jvm/
ENV ACME_LOGS /acme/logs/
RUN mkdir -p "$JVM_PATH" "$ACME_LOGS"
RUN ls -l /acme/
3. Build the image
4. Run the container from that image

**Describe the results you received:**

When building the image : 

Step 15 : RUN mkdir -p "$JVM_PATH" "$ACME_LOGS" 
 ---> Running in a0c973eeaf90
 ---> 4037f4b3c663
Removing intermediate container a0c973eeaf90
Step 16 : RUN ls -l /acme/
/acme/jvm
/acme/logs
Removing intermediate container 43ad3322ba4a

After running the container: 
1. docker exec -it container_name bash
ls -l /acme 
total 0

**Describe the results you expected:**
I expected that the 2 sub directories are created but only the parent one is created

**Additional information you deem important (e.g. issue happens only occasionally):**

docker run cmd: 
docker run -it -d -p 81:8080 -p 443:443 --privileged --restart=always --name integrit -v /acme/docker/volumes/integrit/:/acme/ acme/tomcat:7.0.70

**Comment by espritengineer** [comment]
After debugging I found that when running the container without mounting a host volume ( i.e without -v /acme/docker/volumes/integrit/:/acme/ ) the sub-folders are created. 

I think that creating sub-folders does not apply when mounting a host directory. The only workaround I found is that I created the directory and their sub-directories inside the host and then mount it with a container volume. (Of course in the Dockerfile, the RUN mkdir commands are useless then)
